python/cm/variant.py

```diff
@@ -0,0 +1,319 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cm.logger import log
+from cm.errors import AdcmEx
+from cm.errors import raise_AdcmEx as err
+from cm.models import Prototype, ClusterObject, ServiceComponent, HostComponent, Host
+
+
+def get_cluster(obj):
+    if obj.prototype.type == 'service':
+        cluster = obj.cluster
+    elif obj.prototype.type == 'host':
+        cluster = obj.cluster
+    elif obj.prototype.type == 'cluster':
+        cluster = obj
+    else:
+        return None
+    return cluster
+
+
+def variant_service_in_cluster(obj, args=None):
+    out = []
+    cluster = get_cluster(obj)
+    if not cluster:
```

Comment:
```suggestion
    if cluster is None:
```
I think it would be better to explicitly check for None, since the `get_cluster()` function returns an object or None